pkg/controllers/machine_controller.go

```diff
@@ -888,3 +923,11 @@ func machineDomain(machineID string) libvirt.Domain {
 		UUID: libvirtutils.UUIDStringToBytes(machineID),
 	}
 }
+
+func getShutdownMode(agent *api.GuestAgentStatus) libvirt.DomainShutdownFlagValues {
```

Comment:
What do you think about inlining this few lines?

Reply:
theoretically, i prefer more function for simpler extending but it this case no problem.
If you want to simplify it, we can use automode and libvirt will choose best option alone, but i am afraid of unexpected behavior in this case and logic will depend on another component and behavior can change base on version what can bring more problem for us. So i don't recommend use automode.

Reply:
fixed